api/types.proto

```diff
@@ -696,3 +696,9 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+message Device {
+	string source = 1;
+	string target = 2;
+	string permissions = 3;
```

Comment:
@flx42 This is good information.

> 1) We should allow devices in services, because a services might need a device, any device, no questions asked. It will be the responsibility of the user to make sure the device exists on all nodes.
2) We can start a longer discussion regarding supporting additional resource types (not only GPUs) in Swarm.

I think this is a good plan, but let's make sure we understand the shape of 2 before only going with 1.

The existence of a `Device` message is probably sufficient, since we could add fields to reference a type of device or a set of devices that make up a "class".

cc @crosbymichael